conan/tools/microsoft/msbuild.py

```diff
@@ -16,10 +22,18 @@ def msbuild_arch(arch):
             'armv8': 'ARM64'}.get(str(arch))
 
 
+def msbuild_arch_to_conf_arch(arch):
+    return {'x86': 'Win32',
```

Comment:
I'm a little bit surprised of this difference. In conan-center, recipes using legacy MSBuild were always overriding x86 to pass Win32. Is it not a mistake?